core/bin/zksync_tee_prover/src/tee_prover.rs

```diff
@@ -79,7 +93,7 @@ impl TeeProver {
                 let batch_number = verification_result.batch_number;
                 let msg_to_sign = Message::from_slice(root_hash_bytes)
                     .map_err(|e| TeeProverError::Verification(e.into()))?;
-                let signature = self.config.signing_key.sign_ecdsa(msg_to_sign);
+                let signature = TeeProver::sign_message(&self.config.signing_key, msg_to_sign)?;
```

Comment:
How about `self.sign_message()` which then has access to `self.config.signing_key`??